packages/@lwc/engine-core/src/framework/modules/computed-class-attr.ts

```diff
@@ -59,16 +59,16 @@ function getMapFromClassName(className: string | undefined): Record<string, bool
 }
 
 export function patchClassAttribute(
-    oldVnode: VBaseElement | null,
-    vnode: VBaseElement,
+    oldNode: VBaseElement | VStaticPart | null,
+    node: VBaseElement | VStaticPart,
     renderer: RendererAPI
 ) {
     const {
         elm,
         data: { className: newClass },
-    } = vnode;
+    } = node;
 
-    const oldClass = isNull(oldVnode) ? undefined : oldVnode.data.className;
+    const oldClass = isNull(oldNode) ? undefined : oldNode.data.className;
```

Comment:
```suggestion
    const oldClass = oldNode?.data.className;
```
nit: equivalent using modern syntax

Reply:
In theory `isNull` is more efficient than `?.` because you're only asking the JS engine to evaluate whether the variable is null or not, as opposed to "null or undefined," but who knows if it makes a difference in benchmarks or not.

Reply:
Just to add on, in terms of vnodes, what I've observed is `null` has a special meaning, specially not to render anything.

There are parts of the code base where we explicitly [look for `null` instead of `undefined`](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/engine-core/src/framework/api.ts#L228), I think we might want to do an audit soon to make sure we're being consistent and also rethink wether we want to keep this semantic.